rsrcdump/packutils.py

```diff
@@ -1,48 +1,50 @@
+from typing import Tuple, Union
+
+from io import BytesIO
 import struct
 
 class Unpacker:
-    def __init__(self, data, offset=0):
+    def __init__(self, data: bytes, offset: int=0) -> None:
         self.data = data
-        self.offset = 0
-    
-    def unpack(self, fmt):
+        self.offset = offset
+
+    def unpack(self, fmt: str) -> Tuple:
         record_length = struct.calcsize(fmt)
         fields = struct.unpack_from(fmt, self.data, self.offset)
         self.offset += record_length
         return fields
 
-    def seek(self, offset):
+    def seek(self, offset: int) -> None:
         self.offset = offset
 
-    def read(self, size):
+    def read(self, size: int) -> bytes:
         data_slice = self.data[self.offset : self.offset + size]
         assert len(data_slice) == size
         self.offset += size
         return data_slice
 
-    def unpack_pstr(self, decode=True):
+    def unpack_pstr(self, decode: bool=True) -> Union[str, bytes]:
```

Comment:
Since I'm open to bumping the Python version floor, how about this?
```suggestion
    def unpack_pstr(self, decode: bool=True) -> str | bytes:
```